### PR TITLE
LMS-2149: use subquery to remove items which have already been sent

### PR DESCRIPTION
**Pull request by mlynn-lp**
**Description**
- Only display records which have not been sent in the historical report view.

**Related Issues**
- #2149

**PR Type**
- Enhancement

### CODE REVIEW

**Comment by stephen-oh** [comment]
LMS-2149 is the wrong ticket number, can you update this? I have some other comments incoming.

File: report.php [hunk 1]
```diff
@@ -53,6 +53,7 @@
 $basetable = XAPI_REPORT_SOURCE_FAILED;
 $extraselect = 'x.errortype, x.response';
 $pagename = 'logstorexapierrorlog';
+$subquery = '';
```
Comment (stephen-oh) [inline report.php:56]: remove this, see my comment below

File: report.php [hunk 2]
```diff
@@ -64,6 +65,7 @@
         $basetable = XAPI_REPORT_SOURCE_HISTORICAL;
         $extraselect = 'u.username, x.contextid';
         $pagename = 'logstorexapihistoriclog';
+        $subquery = "x.id NOT IN (SELECT logstorestandardlogid FROM {logstore_xapi_sent_log})";
```
Comment (stephen-oh) [inline report.php:68]: remove this line

File: report.php [hunk 3]
```diff
@@ -120,6 +122,9 @@
 $params = array_merge($params, $inparams);
 
 $where = implode(' AND ', $where);
+if ($id == XAPI_REPORT_ID_HISTORIC) {
+    $where .= ' AND '.$subquery;
+}
```
Comment (stephen-oh) [inline report.php:127]: You could do the following instead, much neater
```
if ($id == XAPI_REPORT_ID_HISTORIC) {
    $where[] = 'x.id NOT IN (SELECT logstorestandardlogid FROM {logstore_xapi_sent_log})';
}

$where = implode(' AND ', $where);
```

**Review by stephen-oh — CHANGES_REQUESTED** [review]

File: report.php (hunk 3 above)
Comment (mlynn-lp) [reply report.php:127]: I have done this on the other branch.

File: report.php (hunk 2 above)
Comment (mlynn-lp) [reply report.php:68]: I have done this on the other branch.

**Comment by mlynn-lp** [comment]
This work was done in error on this branch. The work has been done in LMS-2174

The PR which replaces this is #717

**Pull request closed without merging**